datafusion/physical-expr/src/aggregate/array_agg_distinct.rs

```diff
@@ -181,47 +187,55 @@ mod tests {
     use arrow_array::Array;
     use arrow_array::ListArray;
     use arrow_buffer::OffsetBuffer;
-    use datafusion_common::utils::array_into_list_array;
     use datafusion_common::{internal_err, DataFusionError};
 
-    // arrow::compute::sort cann't sort ListArray directly, so we need to sort the inner primitive array and wrap it back into ListArray.
-    fn sort_list_inner(arr: ScalarValue) -> ScalarValue {
-        let arr = match arr {
-            ScalarValue::List(arr) => arr.value(0),
-            _ => {
-                panic!("Expected ScalarValue::List, got {:?}", arr)
-            }
-        };
+    // arrow::compute::sort can't sort nested ListArray directly, so we compare the scalar values pair-wise.
```

Comment:
What do you think about filing a ticket upstream in arrow-rs to properly support sorting nested lists?

It seems like it would be valuable eventually and could likely be done much more efficiently than using `ScalarValue`

Reply:
Agree, let me file the ticket. But, I think it is not so helpful for this test (values are little). Maybe support it if there is somewhere that needs sorting many nested lists.

Reply:
> Agree, let me file the ticket. But, I think it is not so helpful for this test (values are little). Maybe support it if there is somewhere that needs sorting many nested lists.

Actually, we can just support `array_sort` for nested array, this also need upstream fix